src/services/createRouter.ts

```diff
@@ -321,7 +348,10 @@ export function createRouter<const TRoutes extends Routes, const TOptions extend
     onAfterRouteLeave,
     prefetch: options?.prefetch,
     start,
+    stop,
   }
 
+  plugins.forEach(plugin => addRouterPluginHooks(router, plugin))
+
```

Comment:
for other plugin things like routes, rejections we just kind of spread the plugin with the user args when creating the thing. Could we could do that for hooks as well? Instead of this, maybe pass the plugin to [createRouterHooks](https://github.com/kitbagjs/router/blob/plugins/src/services/createRouterHooks.ts#L17)?

Reply:
Discussed and we are going to follow this PR up with some refactoring around global hooks so that we can accept hooks as part of create router.